lib/auth/user_repository.dart

```diff
@@ -0,0 +1,8 @@
+import 'package:chat_app_front/auth/user.dart';
+import 'package:dartz/dartz.dart';
+
+import '../common/failure.dart';
+
+abstract class UserRepository {
+  Future<String>signUpUser(User user);
```

Comment:
Metoda powinna zwracać jakąś klasę/enum informująca o statusie akcji, nie string z wiadomością. To jest mieszanie logiki i prezentacji